src/utils.js

```diff
@@ -422,7 +422,7 @@ function resolveDependencies(schema, definitions, formData) {
   // Process dependencies updating the local schema properties as appropriate.
   for (const dependencyKey in dependencies) {
     // Skip this dependency if its trigger property is not present.
-    if (formData[dependencyKey] === undefined) {
+    if (!formData.hasOwnProperty(dependencyKey)) {
```

Comment:
@glasserc Unfortunately it looks like this change broke properties and schema dependencies. With the change conditionally required properties are always marked as required (with an asterisk (*)) and conditionally declared properties are always shown. See the playground for the examples.

Note that the dynamic schema still works correctly with the specific `oneOf` case.

Reply:
I'm surprised this passed the tests... I must not have written very good ones. 😄

Reply:
Yeah, that's unfortunate.. the way we `computeDefaults` fills missing fields with `undefined`s. I don't like that design but it's outside the scope of this PR. Thanks for letting me know that I broke it :)